### PR TITLE
Fix Codacy findings and enhance Docker test scripts

### PR DESCRIPTION
**Pull request by egarcia74**
# Pull Request

## 📝 Description

Follow-up cleanup for Docker test helpers and related unit tests. This PR addresses Codacy/review findings, improves command parsing edge-case handling, keeps Docker env resolution aligned with the repo’s `SQL_SERVER_*` convention, and removes a few small sources of review churn without changing production server behavior.

## 🔗 Related Issue

Relates to #658

## 🧪 Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [x] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvements
- [x] 🧹 Code cleanup
- [ ] 🔒 Security improvement
- [ ] 🎨 Style/UI changes
- [ ] 🔨 DevOps/Build changes

### Added

- Focused unit coverage for Docker command helper behavior, including empty quoted argument parsing.

### Changed

- Switched Docker command helper safe-path construction to use `path.delimiter`.
- Updated Docker DB init config to prefer `SQL_SERVER_*` env vars before `DB_*` fallbacks.
- Replaced the non-blocking timeout in the Docker stress test log check with a real synchronous wait.
- Simplified the streaming handler test mock constructor.
- Removed a stale `.gitignore` entry for a tracked Codacy instruction file.

### Deprecated

- None.

### Removed

- Redundant mock-constructor branching in streaming handler tests.
- Stale ignore rule for `.github/instructions/codacy.instructions.md`.

### Fixed

- Empty quoted argument handling in `parseCommandArguments`.
- Review/Codacy findings in Docker test helpers and unit tests.
- Race-prone container log timing check in the Docker stress test.
- Docker env precedence mismatch for `init-db-node.js`.

### Security

- Preserved explicit safe PATH enforcement and non-shell command execution in Docker helper code.

### Test Coverage

- [ ] Unit tests pass (`npm test`)
- [ ] Integration tests pass (if applicable)
- [x] Manual testing completed
- [ ] Performance testing (if applicable)

### Test Details

**Tested on:**

- [ ] Windows
- [x] macOS
- [ ] Linux

**SQL Server versions tested:**

- [ ] SQL Server 2016
- [ ] SQL Server 2019
- [ ] SQL Server 2022

**MCP Clients tested:**

- [ ] Warp
- [ ] Claude Desktop
- [x] Other: N/A for this test/helper-focused PR

### Test Scenarios

1. Ran focused unit tests for `test/unit/docker-command-utils.test.js` and `test/unit/streaming-handler.test.js`.
2. Ran ESLint on the touched helper and test files.
3. Verified `.github/instructions/codacy.instructions.md` is no longer matched by `.gitignore`.

### New Dependencies

- [x] No new dependencies
- [ ] Production dependencies added: N/A
- [ ] Development dependencies added: N/A

### Dependency Changes

- [ ] Dependencies updated
- [ ] Dependencies removed
- [ ] Security vulnerabilities resolved

## 📖 Documentation

- [ ] README.md updated
- [x] Code comments added/updated
- [ ] API documentation updated (if applicable)
- [ ] CHANGELOG.md updated (if applicable)

## ⚠️ Breaking Changes

**What breaks:**

- None.

**Migration path:**

- None.

**Affected users:**

- None.

## 🔍 Code Quality

- [x] Code follows the project's coding standards
- [x] Self-review of code completed
- [x] Code is self-documenting or well-commented
- [ ] No console.log or debug statements left in code
- [x] Error handling is comprehensive
- [x] Security considerations addressed

## 📊 Performance Impact

- [x] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable
- [ ] Bundle size impact: N/A

## 🔒 Security Considerations

- [x] No security implications
- [x] Security review completed
- [ ] Input validation added/updated
- [ ] Authentication/authorization changes tested
- [x] No sensitive data exposed in logs

## 📱 Screenshots/Videos

**Before:**

N/A

**After:**

N/A

## 🤝 Review Checklist

- [ ] Code review completed
- [x] Tests reviewed and adequate
- [ ] Documentation reviewed
- [x] Security implications considered
- [x] Performance impact acceptable
- [x] Breaking changes properly documented

## 📋 Deployment Notes

- [x] No special deployment requirements
- [ ] Database migrations required
- [ ] Configuration changes needed
- [ ] Environment variables added/changed

## 🎯 Next Steps

- [x] No follow-up required
- [ ] Follow-up issue created: N/A
- [ ] Documentation updates needed
- [ ] User communication required

---

## 📝 Additional Notes

This PR is intentionally narrow and limited to test/helper code and review-driven cleanup. No runtime MCP server behavior or public API behavior is intended to change.

If helpful for reviewers: this branch is the clean follow-up replacement for the earlier incremental PR history.

---

**By submitting this PR, I confirm that:**

- [x] I have read and followed the contributing guidelines
- [x] My code follows the project's coding standards
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: test/docker/command-utils.js [hunk 1]
```diff
@@ -4,27 +4,39 @@ import path from 'node:path';
 const NODE_BIN = path.dirname(process.execPath);
 
 // Safe path for execSync to satisfy Sonar S4036
-export const SAFE_PATH = `${NODE_BIN}:/usr/bin:/usr/local/bin:/usr/sbin:/usr/local/sbin:/bin:/sbin`;
+export const SAFE_PATH = [
```
[the remark this answers is not in the document]
Comment (egarcia74) [reply test/docker/command-utils.js:7]: Codacy can you take another look - I accidentally entered an empty template in the PR but have since updated it.

this script is intentionally macOS/Linux-only
